### PR TITLE
fix: fix User-Agent header overriding the passed one

### PR DESCRIPTION
**Pull request by hebertjulio**
🚨Please review the [guidelines for contributing](https://github.com/schemathesis/schemathesis/blob/master/CONTRIBUTING.rst) to this repository.

### Description
Fix User-Agent header overriding the passed one.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Added a changelog entry
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/schemathesis/loaders.py [hunk 1]
```diff
@@ -259,7 +265,10 @@ def from_asgi(
     validate_schema: bool = True,
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
-    kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if "headers" not in kwargs or kwargs["headers"] is None:
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:268]: I would not expect `None` here and keep the previous behavior:

```python
headers = kwargs.setdefault("headers", {})
```

File: src/schemathesis/loaders.py [hunk 2]
```diff
@@ -259,7 +265,10 @@ def from_asgi(
...
+        kwargs["headers"] = {}
+    if "user-agent" not in set(k.lower() for k in kwargs["headers"]):
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:270]:
```suggestion
    if "user-agent" not in {header.lower() for header in kwargs["headers"]}:
```

File: src/schemathesis/runner/impl/core.py [hunk 3]
```diff
@@ -238,7 +238,8 @@ def network_test(
     """A single test body will be executed against the target."""
     # pylint: disable=too-many-arguments
     headers = headers or {}
-    headers.setdefault("User-Agent", USER_AGENT)
+    if "user-agent" not in set(k.lower() for k in headers):
```
Comment (Stranger6667) [inline src/schemathesis/runner/impl/core.py:241]:
```suggestion
    if "user-agent" not in {header.lower() for header in headers}:
```

File: src/schemathesis/loaders.py [hunk 4]
```diff
@@ -192,7 +195,10 @@ def from_wsgi(
...
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:200]:
```suggestion
    if "user-agent" not in {header.lower() for header in kwargs["headers"]}:
```

File: src/schemathesis/loaders.py [hunk 5]
```diff
@@ -60,7 +60,10 @@ def from_uri(
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
     """Load a remote resource and parse to schema instance."""
-    kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if "headers" not in kwargs or kwargs["headers"] is None:
+        kwargs["headers"] = {}
+    if "user-agent" not in set(k.lower() for k in kwargs["headers"]):
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:65]:
```suggestion
    if "user-agent" not in {header.lower() for header in kwargs["headers"]}:
```

File: src/schemathesis/runner/impl/core.py [hunk 6]
```diff
@@ -356,7 +357,8 @@ def _prepare_wsgi_headers(
     headers: Optional[Dict[str, Any]], auth: Optional[RawAuth], auth_type: Optional[str]
 ) -> Dict[str, Any]:
     headers = headers or {}
-    headers.setdefault("User-Agent", USER_AGENT)
+    if "user-agent" not in set(k.lower() for k in headers):
```
Comment (Stranger6667) [inline src/schemathesis/runner/impl/core.py:360]:
```suggestion
    if "user-agent" not in {header.lower() for header in headers}:
```

File: src/schemathesis/loaders.py [hunk 7]
```diff
@@ -192,7 +195,10 @@ def from_wsgi(
     validate_schema: bool = True,
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
-    kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if "headers" not in kwargs or kwargs["headers"] is None:
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:198]: I would not expect `None` here and keep the previous behavior:

```python
headers = kwargs.setdefault("headers", {})
```

File: src/schemathesis/loaders.py [hunk 8]
```diff
@@ -60,7 +60,10 @@ def from_uri(
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
     """Load a remote resource and parse to schema instance."""
-    kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if "headers" not in kwargs or kwargs["headers"] is None:
```
Comment (Stranger6667) [inline src/schemathesis/loaders.py:63]: I would not expect `None` here and keep the previous behavior:

```python
headers = kwargs.setdefault("headers", {})
```

File: docs/changelog.rst [hunk 9]
```diff
@@ -1374,6 +1375,7 @@ Deprecated
 .. _0.3.0: https://github.com/schemathesis/schemathesis/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/schemathesis/schemathesis/compare/v0.1.0...v0.2.0
 
+.. _#757: https://github.com/schemathesis/schemathesis/issues/757
```
Comment (Stranger6667) [inline docs/changelog.rst:1378]: Could you please move this line one line below to keep the entries ordered?

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
Thanks! I left a couple of comments, but basically, there are two things - using `setdefault` and set literal.

Otherwise, please implement this behavior also for the `Case._get_headers` method and add a test case to [this test](https://github.com/schemathesis/schemathesis/blob/master/test/test_models.py#L52)

**Comment by Stranger6667** [comment]
Can you please, use `git rebase`? commits you are pushing looks like `git cherry-pick` and imply some noise when I review them

**Comment by hebertjulio** [comment]
Sorry @Stranger6667 , i'm using git rebase, but i must be using it the wrong way. Below follows the step by step that I'm doing.

git checkout master
git fetch upstream
git rebase upstream/master
git checkout fixbranch
git rebase master

**Comment by Stranger6667** [comment]
hm it is strange :) I will take a look at this workflow later today

**Pull request closed without merging**